magenta-lib/src/main/scala/magenta/tasks/FastlyComputeTasks.scala

```diff
@@ -27,34 +27,49 @@ case class FastlyComputeTasks(s3Package: S3Path)(implicit
       resources: DeploymentResources,
       stopFlag: => Boolean
   ): Unit = {
-    FastlyApiClientProvider.get(keyRing).foreach { client =>
-      val activeVersionNumber =
-        getActiveVersionNumber(client, resources.reporter, stopFlag)
-      val nextVersionNumber =
-        clone(activeVersionNumber, client, resources.reporter, stopFlag)
-
-      uploadPackage(
-        nextVersionNumber,
-        s3Package,
-        client,
-        resources.reporter,
-        stopFlag
-      )
+    FastlyApiClientProvider.get(keyRing) match {
+      case None =>
+        resources.reporter.fail("Failed to fetch Fastly API credentials")
```

Comment:
Might be worth expanding this slightly to give a clearer sense of where credentials should exist.